debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -418,17 +429,21 @@ protected void generateDeleteRecord(TableId tableId, Object[] oldRowData, Blocki
         Map<String, ?> offset = sourceInfo.offset();
         String topicName = topicSelector().topicNameFor(tableId);
         Envelope envelope = tableSchema.getEnvelopeSchema();
+        Struct source = sourceInfo.source();
 
         // create the regular delete record
         ChangeEvent changeEvent = new ChangeEvent(
                 new SourceRecord(
                         partition, offset, topicName, null,
                         keySchema, key, envelope.schema(),
-                        envelope.delete(value, sourceInfo.source(), clock().currentTimeInMillis())),
+                        envelope.delete(value, source, clock().currentTimeInMillis())),
                 lastCompletelyProcessedLsn);
         if (logger.isDebugEnabled()) {
             logger.debug("sending delete event '{}' to topic '{}'", changeEvent.getRecord(), topicName);
         }
+        if (enhancedDebug) {
+            logger.info("[RECORD_DEBUG] Sending delete record {} with source {} for topic {} at lsn {}", changeEvent.getRecord(), source, topicName, lastCompletelyProcessedLsn);
```

Comment:
Were you able to trigger this log message locally? Do you know if logging Struct types in this way works? I'm concerned it might just log a reference to the object rather than the actual value.